Neos.Flow/Classes/ResourceManagement/ResourceManager.php

```diff
@@ -151,30 +153,21 @@ protected function initialize()
      */
     public function importResource($source, $collectionName = ResourceManager::DEFAULT_PERSISTENT_COLLECTION_NAME, $forcedPersistenceObjectIdentifier = null)
     {
-        $this->initialize();
-        if (!isset($this->collections[$collectionName])) {
-            throw new Exception(sprintf('Tried to import a file into the resource collection "%s" but no such collection exists. Please check your settings and the code which triggered the import.', $collectionName), 1375196643);
-        }
-
-        /* @var CollectionInterface $collection */
-        $collection = $this->collections[$collectionName];
-
-        try {
-            $resource = $collection->importResource($source);
-            if ($forcedPersistenceObjectIdentifier !== null) {
-                ObjectAccess::setProperty($resource, 'Persistence_Object_Identifier', $forcedPersistenceObjectIdentifier, true);
+        if (is_resource($source)) {
+            $content = stream_get_contents($source);
```

Comment:
But this will load all resources fully into the memory, right? Wouldn't this have a huge impact for large uploads like videos?

Reply:
That somehow is the price for sanitizing ...